samtranslator/parser/parser.py

```diff
@@ -41,16 +46,15 @@ def validate_datatypes(sam_template):  # type: ignore[no-untyped-def]
             # NOTE: Properties isn't required for SimpleTable, so we can't check
             # `not isinstance(sam_resources.get("Properties"), dict)` as this would be a breaking change.
             # sam_resource.properties defaults to {} in SamTemplate init
-            if not isinstance(sam_resource.properties, dict):
-                raise InvalidDocumentException(
-                    [
-                        InvalidResourceException(
-                            resource_logical_id,
-                            "All 'Resources' must be Objects and have a 'Properties' Object. If "
-                            "you're using YAML, this may be an indentation issue.",
-                        )
-                    ]
+            try:
+                sam_expect(
+                    sam_resource.properties, resource_logical_id, "Properties", is_resource_attribute=True
+                ).to_be_a_map(
+                    "All 'Resources' must be Objects and have a 'Properties' Object. If "
+                    "you're using YAML, this may be an indentation issue."
```

Comment:
This is too verbose IMO. Can we remove this custom message and leave it default?